In [4]:
import pandas as pd

pd.set_option('display.max_rows', 30)

### Loading the data dictionary from the file

In [5]:
# loading the data dictionary
file_name = 'institution_data_dictionary.csv'
df_data_dict_inst = pd.read_csv(file_name)
df_data_dict_inst.shape

(3251, 11)

### This shows that the data dictionary has 3251 rows. We can consider those as the features in our dataet.

In [35]:
# Displaying some sample values of the data dictionary
df_data_dict_inst

NAME OF DATA ELEMENT dev-category  \
0                               Unit ID for institution         root   
1                        8-digit OPE ID for institution         root   
2                        6-digit OPE ID for institution         root   
3                                      Institution name       school   
4                                                  City       school   
...                                                 ...          ...   
3246  Median earnings of students working and not en...     earnings   
3247  Median earnings of independent students workin...     earnings   
3248  Median earnings of dependent students working ...     earnings   
3249  Median earnings of non-male students working a...     earnings   
3250  Median earnings of male students working and n...     earnings   

                                developer-friendly name API data type  \
0                                                    id       integer   
1                                               ope8_id        string   
2                                               ope6_id        string   
3                                                  name  autocomplete   
4                                                  city  autocomplete   
...                                                 ...           ...   
3246  10_yrs_after_entry.median_earnings.highest_ter...       integer   
3247  10_yrs_after_entry.median_earnings.independent...       integer   
3248  10_yrs_after_entry.median_earnings.dependent_s...       integer   
3249  10_yrs_after_entry.median_earnings.non_male_st...       integer   
3250   10_yrs_after_entry.median_earnings.male_students       integer   

             INDEX           VARIABLE NAME VALUE LABEL    SOURCE  \
0              NaN                  UNITID   NaN   NaN     IPEDS   
1      varchar(10)                   OPEID   NaN   NaN     IPEDS   
2      varchar(10)                  OPEID6   NaN   NaN     IPEDS   
3         fulltext                  INSTNM   NaN   NaN     IPEDS   
4     varchar(200)                    CITY   NaN   NaN     IPEDS   
...            ...                     ...   ...   ...       ...   
3246           NaN    MD_EARN_WNE_INC3_P10   NaN   NaN  Treasury   
3247           NaN  MD_EARN_WNE_INDEP1_P10   NaN   NaN  Treasury   
3248           NaN  MD_EARN_WNE_INDEP0_P10   NaN   NaN  Treasury   
3249           NaN   MD_EARN_WNE_MALE0_P10   NaN   NaN  Treasury   
3250           NaN   MD_EARN_WNE_MALE1_P10   NaN   NaN  Treasury   

     SHOWN/USE ON SITE NOTES  
0                  Yes   NaN  
1                  Yes   NaN  
2                  Yes   NaN  
3                  Yes   NaN  
4                  Yes   NaN  
...                ...   ...  
3246               NaN   NaN  
3247               NaN   NaN  
3248               NaN   NaN  
3249               NaN   NaN  
3250               NaN   NaN  

[3251 rows x 11 columns]

In [37]:
### Getting to know which variables to pick for feature selection
df_data_dict_inst[['NAME OF DATA ELEMENT','VARIABLE NAME','VALUE','LABEL']][0:10]

NAME OF DATA ELEMENT VARIABLE NAME VALUE LABEL
0                     Unit ID for institution        UNITID   NaN   NaN
1              8-digit OPE ID for institution         OPEID   NaN   NaN
2              6-digit OPE ID for institution        OPEID6   NaN   NaN
3                            Institution name        INSTNM   NaN   NaN
4                                        City          CITY   NaN   NaN
5                              State postcode        STABBR   NaN   NaN
6                                    ZIP code           ZIP   NaN   NaN
7                  Accreditor for institution  ACCREDAGENCY   NaN   NaN
8              URL for institution's homepage       INSTURL   NaN   NaN
9  URL for institution's net price calculator        NPCURL   NaN   NaN

In [ ]:
print

In [33]:
list_of_features=['INSTNM','INSTURL']

In [30]:
# Load the most recent data file with the instituion names
data_file_inst = 'dataset/CollegeScorecard_Raw_Data_03142022/Most-Recent-Cohorts-Institution.csv'
df_inst = pd.read_csv(data_file_inst, low_memory=False)

In [31]:
df_inst.shape

(6694, 2989)

In [34]:
df_features_inst = df_inst[list_of_features]
df_features_inst

INSTNM                        INSTURL
0                     Alabama A & M University                  www.aamu.edu/
1          University of Alabama at Birmingham           https://www.uab.edu/
2                           Amridge University     www.amridgeuniversity.edu/
3          University of Alabama in Huntsville                   www.uah.edu/
4                     Alabama State University                 www.alasu.edu/
...                                        ...                            ...
6689        Georgia Military College - Eastman  https://www.gmc.edu/index.cms
6690  American College of Barbering - Florence                            NaN
6691      HCI College - Fort Lauderdale Campus                   www.hci.edu/
6692          ABC Adult School - Cabrillo Lane                            NaN
6693           Urban Barber College - San Jose    www.urbanbarbercollege.com/

[6694 rows x 2 columns]